documentation/istio_keycloak_integration.asciidoc

```diff
@@ -0,0 +1,213 @@
+:toc: macro
+toc::[]
+:idprefix:
+:idseparator: -
+== Authentication with Keycloak and Istio
+
+=== Introducing Keycloak
+https://www.keycloak.org/[Keycloak] is an opensource IAM (Identity and Access Management) solution with a broad set of features like SSO, authentication and authorization, social login, multifactor authentication etc.
```

Comment:
from our experience in devon4j structuring documentation like this does not work out and is not maintainable.
Therefore my suggestion is:
* we create separate asciidoc files for keycloak and istio where we can document general aspects for each of them
* in file like this where we want to document the integration between those two we can link to each of them but shall IMHO not explain anything more but focus on the integration aspects.